Extend kubernetes interrelation variables in nginx.tmpl

distol opened the pull request:
This PR improves variables, that allows associate (in logs and in lua) nginx servers and locations with related kubernetes resources:
1. According to TCP/IP (and common sense), `$service_name` is not enough to uniquely identify "service", we need `$service_port` for that.
2. When you define rules in ingress resource, you use path. So it would be very useful to be able to use the same path in logs (and in lua).

aledbf commented:
@distol please sign the CLA

antoineco commented on rootfs/etc/nginx/template/nginx.tmpl:
```diff
@@ -834,6 +836,7 @@ stream {
             set $namespace      "{{ $ing.Namespace }}";
             set $ingress_name   "{{ $ing.Rule }}";
             set $service_name   "{{ $ing.Service }}";
+            set $service_port   "{{ if $location.Port }}{{ $location.Port }}{{ end }}";
```
The conditional is not necessary here.

antoineco commented on rootfs/etc/nginx/template/nginx.tmpl:
```diff
@@ -829,6 +829,12 @@ stream {
 
         location {{ $path }} {
+            {{ $ing := (getIngressInformation $location.Ingress $location.Path) }}
+            {{/* $ing.Metadata contains the Ingress metadata */}}
```
Outdated comment ($ing.Metadata is not referenced).

antoineco commented on rootfs/etc/nginx/template/nginx.tmpl:
```diff
@@ -829,6 +831,14 @@ stream {
+            set $namespace      "{{ $ing.Namespace }}";
+            set $ingress_name   "{{ $ing.Rule }}";
+            set $service_name   "{{ $ing.Service }}";
+            set $service_port   "{{ if $location.Port }}{{ $location.Port }}{{ end }}";
```
The conditional is not necessary here.

antoineco commented:
Oh, you also have to run `make code-generator` by the way :) It creates a gzipped bytes array of the `nginx.tmpl` file which gets unpacked in memory during tests.

antoineco commented:
/lgtm

aledbf commented:
@distol please squash the commits and we are ready to merge

distol commented:
@aledbf, commits are already squashed. I've made 3 small independent changes, so there are 3 commits, one for each change with its own comment.

aledbf commented:
@distol friendly ping. Please rebase and squash

ElvinEfendi commented on rootfs/etc/nginx/template/nginx.tmpl:
```diff
@@ -829,6 +831,13 @@ stream {
+            set $ingress_name   "{{ $ing.Rule }}";
+            set $service_name   "{{ $ing.Service }}";
+            set $service_port   "{{ $location.Port }}";
+            set $location_path  "{{ $location.Path }}";
```
Could you describe a use case for this? Are `$request` or `$request_uri` not enough to identify which path got hit?

distol commented:
> @distol friendly ping. Please rebase and squash

@aledbf, I've already written, that commits have been squashed from the very beginning.  I've made 3 small independent changes, and to my sense, it should be 3 commits, one for each change with its own comment. And same time it looks not reasonable to make three pull requests because changes are too small.

Please have a look at commit log and if after that you will not change your mind — no problem, I'll squash these three changes to one commit.

P.S. Rebased to the master and removed bindata.

aledbf commented:
/lgtm
/approve

aledbf commented:
@distol thanks!

aledbf commented:
@distol next time, please squash the commits
